contracts/modules/licensing/LicensingModuleV2.sol

```diff
@@ -0,0 +1,360 @@
+    GovernableUpgradeable,
+    UUPSUpgradeable
+{
+    using ERC165Checker for address;
+    using IPAccountChecker for IIPAccountRegistry;
+    using EnumerableSet for EnumerableSet.UintSet;
+    using EnumerableSet for EnumerableSet.AddressSet;
+    using Strings for *;
+    using IPAccountStorageOps for IIPAccount;
+
+    /// @inheritdoc IModule
+    string public constant override name = LICENSING_MODULE_KEY;
+
+    /// @notice Returns the canonical protocol-wide RoyaltyModule
+    /// @custom:oz-upgrades-unsafe-allow state-variable-immutable
+    RoyaltyModule public immutable ROYALTY_MODULE;
+
+    /// @notice Returns the canonical protocol-wide LicenseRegistry
+    /// @custom:oz-upgrades-unsafe-allow state-variable-immutable
+    ILicenseRegistryV2 public immutable LICENSE_REGISTRY;
+
+    /// @notice Returns the dispute module
+    /// @custom:oz-upgrades-unsafe-allow state-variable-immutable
+    IDisputeModule public immutable DISPUTE_MODULE;
+
+    ILicenseNFT public immutable LICENSE_NFT;
+
+    // keccak256(abi.encode(uint256(keccak256("story-protocol.LicensingModule")) - 1)) & ~bytes32(uint256(0xff));
+    bytes32 private constant LicensingModuleStorageLocation =
+        0x0f7178cb62e4803c52d40f70c08a6f88d6ee1af1838d58e0c83a222a6c3d3100;
+
+    /// Constructor
+    /// @param accessController The address of the AccessController contract
+    /// @param ipAccountRegistry The address of the IPAccountRegistry contract
+    /// @param royaltyModule The address of the RoyaltyModule contract
+    /// @param registry The address of the LicenseRegistry contract
+    /// @param disputeModule The address of the DisputeModule contract
+    /// @custom:oz-upgrades-unsafe-allow constructor
+    constructor(
+        address accessController,
+        address ipAccountRegistry,
+        address royaltyModule,
+        address registry,
+        address disputeModule,
+        address licenseNFT
+    ) AccessControlled(accessController, ipAccountRegistry) {
+        ROYALTY_MODULE = RoyaltyModule(royaltyModule);
+        LICENSE_REGISTRY = ILicenseRegistryV2(registry);
+        DISPUTE_MODULE = IDisputeModule(disputeModule);
+        LICENSE_NFT = ILicenseNFT(licenseNFT);
+        _disableInitializers();
+    }
+
+    /// @notice initializer for this implementation contract
+    /// @param governance The address of the governance contract
+    function initialize(address governance) public initializer {
+        __ReentrancyGuard_init();
+        __UUPSUpgradeable_init();
+        __GovernableUpgradeable_init(governance);
+    }
+
+    function attachLicenseTerms(
+        address ipId,
+        address licenseTemplate,
+        uint256 licenseTermsId
+    ) external verifyPermission(ipId) {
+        _verifyIpNotDisputed(ipId);
+        LICENSE_REGISTRY.attachLicenseTermsToIp(ipId, licenseTemplate, licenseTermsId);
+        emit LicenseTermsAttached(msg.sender, ipId, licenseTemplate, licenseTermsId);
+    }
+
+    function mintLicenseTokens(
```

Comment:
it's very hard to review a completely new code.  Hope there is not much change of the business logic from the V1 code.  need a comparison session and use case test coverage.

Reply:
yes, all behavior should be same as beta.